### DISCUSSION TITLE
flash dfuse over USART

### DISCUSSION THREAD

**Discussion by amfern** [fwupd/fwupd#3648] [Ideas]
**Is your feature request related to a problem? Please describe.**
st microcontrollers are capable of being flash over USART, what would it take to bring this feature to fwupd?
are there any existing plugin i can take as example?

**Describe the solution you'd like**
Flash st microcontrollers over usart

**Additional context**
file:///tmp/dm00589815-usb-dfuusart-protocols-used-in-stm32mp1-series-bootloaders-stmicroelectronics-1.pdf

**Comment by hughsie** [comment]
USART looks like a variant of DFU, just like we have ST and AVR. I think it would fit quite nicely in the existing DFU plugin: https://github.com/fwupd/fwupd/tree/master/plugins/dfu

**Comment by amfern** [comment]
But USART is a serial, while the dfu plugin is using USB.
Is there infrastructure in fwupd to use serial?

**Comment by hughsie** [comment]
Ohh, it's literally talking over a tty? In which case there might be some inspiration from the https://github.com/fwupd/fwupd/blob/master/plugins/altos/fu-altos-device.c plugin instead.
